### PR TITLE
Rename some type names that had "Type"  in them

### PR DESCRIPTION
**Pull request by thockin**
Just a bit of redundancy removal.  Old names left for compat.

/kind cleanup

```release-note
NONE
```
 /sig network

### CODE REVIEW

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -3594,17 +3594,18 @@ const (
 	ServiceTypeExternalName ServiceType = "ExternalName"
 )
 
-// ServiceInternalTrafficPolicyType describes the type of traffic routing for
+// ServiceInternalTrafficPolicy describes the type of traffic routing for
 // internal traffic
-type ServiceInternalTrafficPolicyType string
+type ServiceInternalTrafficPolicy string
+type ServiceInternalTrafficPolicyType = ServiceInternalTrafficPolicy
```
Comment (dcbw) [inline pkg/apis/core/types.go:3600]: @thockin missing the `// for backwards compat` comment here? Or maybe we don't need it?

Comment (dcbw) [inline pkg/apis/core/types.go:3600]: @thockin missing the // for backwards compat comment here? Or maybe we don't need it?

**Comment by khenidak** [comment]
/lgtm

**Q**: do we have rules around dep compatibility? this will break people who are using these constants/enums via go-client, right?

**Comment by thockin** [comment]
I think I have left all the old types as aliases - did I miss some?

**Comment by thockin** [comment]
Updated and rebased, still some oddness

@dims @cheftako for legacy cloud provider

**Comment by thockin** [comment]
@jiahuif @apelisse 

This PR renames some types and adds aliases for the old names.  It seems to break openapi enums.

The 1st commit renames `IPFamilyPolicyType` -> `IPFamilyPolicy`.  It is listed as `+enum` but I see swagger.json does not list it as an enum - just string.  I see this for fields of `IPFamily` type too.  Why is +enum not working?

The 2nd commit renames `ServiceExternalTrafficPolicyType` -> `ServiceExternalTrafficPolicy`.  I see swagger.json CHANGE it from an enum to a string but I have no idea why.

Can you offer insight?  Am I doing something wrong?

**Comment by jiahuif** [comment]
There is no assumption about naming of enum types. The type is treated as an enum type iff the +enum marker presents.

File: api/openapi-spec/v3/api__v1_openapi.json [hunk 2]
```diff
@@ -29011,12 +29011,8 @@
             "type": "string"
           },
           "externalTrafficPolicy": {
-            "description": "externalTrafficPolicy denotes if this Service desires to route external traffic to node-local or cluster-wide endpoints. \"Local\" preserves the client source IP and avoids a second hop for LoadBalancer and Nodeport type services, but risks potentially imbalanced traffic spreading. \"Cluster\" obscures the client source IP and may cause a second hop to another node, but should have good overall load-spreading.\n\nPossible enum values:\n - `\"Cluster\"` specifies node-global (legacy) behavior.\n - `\"Local\"` specifies node-local endpoints behavior.",
-            "type": "string",
-            "enum": [
-              "Cluster",
-              "Local"
-            ]
```
Comment (liggitt) [inline api/openapi-spec/v3/api__v1_openapi.json:29019]: is this correct? surprised to see an externally visible change here

File: api/openapi-spec/swagger.json [hunk 3]
```diff
@@ -10200,11 +10200,7 @@
           "type": "string"
         },
         "externalTrafficPolicy": {
-          "description": "externalTrafficPolicy denotes if this Service desires to route external traffic to node-local or cluster-wide endpoints. \"Local\" preserves the client source IP and avoids a second hop for LoadBalancer and Nodeport type services, but risks potentially imbalanced traffic spreading. \"Cluster\" obscures the client source IP and may cause a second hop to another node, but should have good overall load-spreading.\n\nPossible enum values:\n - `\"Cluster\"` specifies node-global (legacy) behavior.\n - `\"Local\"` specifies node-local endpoints behavior.",
-          "enum": [
-            "Cluster",
-            "Local"
-          ],
```
Comment (liggitt) [inline api/openapi-spec/swagger.json:10207]: same question here... is the externally visible change correct?

**Comment by jiahuif** [comment]
/hold 
/assign
Investigating the enum generation issue.

File: api/openapi-spec/swagger.json (hunk 3 above)
Comment (jiahuif) [reply api/openapi-spec/swagger.json:10207]: No. However, the problem was the constant aliases breaking enum OpenAPI generator. I am now working on a fix.

Comment (jiahuif) [reply api/openapi-spec/swagger.json:10207]: The problem is with `k8s.io/gengo`, trying to fix.

File: staging/src/k8s.io/api/core/v1/types.go [hunk 4]
```diff
@@ -4160,15 +4160,24 @@ const (
...
-	// ServiceExternalTrafficPolicyTypeCluster specifies node-global (legacy) behavior.
-	ServiceExternalTrafficPolicyTypeCluster ServiceExternalTrafficPolicyType = "Cluster"
+	// ServiceExternalTrafficPolicyLocal specifies node-local endpoints behavior.
+	ServiceExternalTrafficPolicyLocal ServiceExternalTrafficPolicy = "Local"
+	// ServiceExternalTrafficPolicyCluster specifies node-global (legacy) behavior.
+	ServiceExternalTrafficPolicyCluster ServiceExternalTrafficPolicy = "Cluster"
+)
+
+// for backwards compat
+type ServiceExternalTrafficPolicyType = ServiceExternalTrafficPolicy
```
Comment (jiahuif) [inline staging/src/k8s.io/api/core/v1/types.go:4179]:
```go
// for backwards compat
// +enum
type ServiceExternalTrafficPolicyType = ServiceExternalTrafficPolicy
```
To workaround a k/gengo bug.

**Comment by spiffxp** [comment]
/milestone v1.24
We are past code freeze and this doesn't appear to be test-related to make it for test freeze.  If there is an exception to allow this through past code freeze, please re-milestone

File: staging/src/k8s.io/api/core/v1/types.go (hunk 4 above)
Comment (thockin) [reply staging/src/k8s.io/api/core/v1/types.go:4179]: Why does `IPFamily` not show up in swagger.json (field name IPFamilies) as an enum?

Comment (thockin) [reply staging/src/k8s.io/api/core/v1/types.go:4181]: I did as you suggest but it doesn't really work:

It half works for 1 case, but duplicates the enum values in the list, and doesn't work at all for the other cases.

**Comment by thockin** [comment]
Obsoleted by https://github.com/kubernetes/kubernetes/pull/110990 and https://github.com/kubernetes/kubernetes/pull/110991

**Pull request closed without merging**